### PR TITLE
Caches dates for select operations

### PR DESCRIPTION
**Pull request by naomijub**
https://github.com/naomijub/wooridb/issues/95

### CODE REVIEW

File: woori-db/src/actors/history.rs [hunk 1]
```diff
@@ -83,9 +83,18 @@ fn get_previous_registry(fractions: Vec<&str>) -> Result<Option<DataRegister>, E
...
     let state = &state[..(state.len() - 1)];
 
-    let resp: Result<DataRegister, Error> = match ron::de::from_str(state) {
+    let resp: Result<(DataRegister, HashMap<String, Types>), Error> = match ron::de::from_str(state)
+    {
         Ok(x) => Ok(x),
         Err(_) => Err(Error::FailedToParseRegistry),
     };
-    Ok(Some(resp?))
+    let resp_insert: Result<DataRegister, Error> = match ron::de::from_str(state) {
```
Comment (otaviopace) [inline woori-db/src/actors/history.rs:91]: On both matches the code could be simplified to use `map_err`, but that is just code simplification/style.

File: woori-db/src/actors/state.rs [hunk 2]
```diff
@@ -75,8 +75,10 @@ impl Handler<PreviousRegistry> for Executor {
                 .ok_or(Error::FailedToParseRegistry)?
                 .to_owned();
             let state = &state[..(state.len() - 1)];
+            println!("{:?}", state);
```
Comment (otaviopace) [inline woori-db/src/actors/state.rs:78]: This `println` should stay here? Was it for debugging?

File: woori-db/src/actors/state.rs [hunk 3]
```diff
@@ -75,8 +75,10 @@ impl Handler<PreviousRegistry> for Executor {
...
 
-            let resp: Result<DataRegister, Error> = match from_str(state) {
+            let resp: Result<(DataRegister, HashMap<String, Types>), Error> = match from_str(state)
+            {
```
Comment (otaviopace) [inline woori-db/src/actors/state.rs:81]: Another `map_err`, again you decide if you want to use it, up to personal preference :slightly_smiling_face:

File: woori-db/src/io/read.rs [hunk 4]
```diff
@@ -137,11 +140,13 @@ pub fn local_data() -> Result<BTreeMap<String, BTreeMap<Uuid, DataRegister>>, er
...
 
-    let data: Result<BTreeMap<String, BTreeMap<Uuid, DataRegister>>, error::Error> =
-        match ron::de::from_str(&s) {
-            Ok(x) => Ok(x),
-            Err(_) => Err(error::Error::FailedToParseState),
-        };
+    let data: Result<
+        BTreeMap<String, BTreeMap<Uuid, (DataRegister, HashMap<String, Types>)>>,
+        error::Error,
+    > = match ron::de::from_str(&s) {
```
Comment (otaviopace) [inline woori-db/src/io/read.rs:146]: `map_err`

File: woori-db/src/actors/history.rs (hunk 1 above)
Comment (naomijub) [reply woori-db/src/actors/history.rs:91]: ron::error doesnt convert to the correct error for every case, so it is needed to use this kind of solution

File: woori-db/src/actors/state.rs (hunk 2 above)
Comment (naomijub) [reply woori-db/src/actors/state.rs]: 2f36dec

File: woori-db/src/controllers/tx.rs [hunk 5]
```diff
@@ -306,7 +306,7 @@ pub async fn insert_controller(
     actor
         .send(CheckForUniqueKeys {
             entity: args.entity.to_owned(),
-            content: encrypted_content,
+            content: encrypted_content.clone(),
```
[the remark this answers is not in the document]
Comment (naomijub) [reply woori-db/src/controllers/tx.rs]: 2f36dec

**Pull request merged**